v2/internal/controllers/reconcile_policy_test.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1alpha1api20200601"
+)
+
+func Test_ReconcilePolicy_SkipReconcileAddedAlongWithTagsChange_ReconcileIsSkipped(t *testing.T) {
+	testModificationSkipped(t, "skip")
+}
+
+func Test_UnknownReconcilePolicy_SkipsAllUpdates(t *testing.T) {
+	testModificationSkipped(t, "UNKNOWN")
+}
```

Comment:
hmm should the reconciliation be skipped if there is an unknown string in the policy?

Reply:
Resolving this and moving discussion into [this comment](https://github.com/Azure/azure-service-operator/pull/2060#discussion_r794972922) which talks about it more in-depth.